salon_creator/lib/screens/salon_creation_screen.dart

```diff
@@ -8,20 +20,529 @@ class SalonCreationScreen extends StatefulWidget {
 }
 
 class _SalonCreationScreenState extends State<SalonCreationScreen> {
+  bool isFilled = false;
+  bool inProgress = false;
+  final ImagePicker imagePicker = ImagePicker();
+  int index = 0;
+  List<XFile> _imageFiles = [];
+
+  ScrollController _scrollController = ScrollController();
+  TextEditingController categoriesController = TextEditingController();
+  TextEditingController descriptionController = TextEditingController();
+  TextEditingController priceController = TextEditingController();
+  TextEditingController salonNameController = TextEditingController();
+  XFile image;
+
+  void _updateScreenContext() {
+    var _hasProfileChanged = salonNameController.text.isNotEmpty &&
+        categoriesController.text.isNotEmpty &&
+        descriptionController.text.isNotEmpty &&
+        priceController.text.isNotEmpty &&
+        _imageFiles.isNotEmpty;
+    print(salonNameController.text.isNotEmpty);
+
+    print(priceController.text.isNotEmpty);
+    print(descriptionController.text.isNotEmpty);
+    print(categoriesController.text.isNotEmpty);
+    print(_imageFiles.isNotEmpty);
+    print(isFilled);
+
+    setState(() {
+      isFilled = _hasProfileChanged;
+    });
+  }
+
+  _scrollToRight() {
+    _scrollController.jumpTo(_scrollController.position.maxScrollExtent);
+  }
+
   @override
   Widget build(BuildContext context) {
+    final mq = MediaQuery.of(context).size;
+    final screenWidth = mq.width;
+    final screenHeight = mq.height;
+
     return Scaffold(
-      appBar: AppBar(
-        actions: [
-          TextButton(
-            onPressed: () {
-              Navigator.of(context).pushReplacementNamed('/home');
-            },
-            child: Text("スキップ"),
+      appBar: inProgress ? null : _buildAppBar(context),
+      body: inProgress
+          ? _buildCreatingProgressScreen()
+          : SafeArea(
+              child: GestureDetector(
+                onTap: () {
+                  FocusScope.of(context).unfocus();
+                },
+                child: SingleChildScrollView(
+                  child: Container(
+                    child: Column(
+                      children: [
+                        SizedBox(
+                          height: 8,
+                        ),
+                        _buildMediaPreviewerDivider(),
+                        SizedBox(
+                          height: 8,
+                        ),
+                        _buildMediaPreviewer(
+                            screenHeight, screenWidth, context),
+                        _buildMediaPane(context),
+                        _buildSalonDetailForms(screenWidth),
+                        isFilled ? Container() : Text("空欄の部分があります"),
```

Comment:
```suggestion
                        isFormFilled ? Container() : Text("空欄の部分があります"),
```
Rename for better readability?